app/src/main/java/com/akp/ceg4110/quickreports/DatabaseAccessor.java

```diff
@@ -0,0 +1,234 @@
+                String selectPicturesQuery = String.format("SELECT %1$s FROM %2$s WHERE name = '%3$s';",
+                        PICTURE_COLUMN, PICTURE_TABLE, name);
+                pictureCursor = db.rawQuery(selectPicturesQuery, null);
+                pictureCursor.moveToFirst();
+                List<Bitmap> images = new ArrayList<Bitmap>();
+                int imageCount = pictureCursor.getCount();
+                //Get pictures from picture table
+                for(int j = 0; j < pictureCursor.getCount(); j++){
+                    byte[] imageBytes = pictureCursor.getBlob(j);
+                    Bitmap bitImage = byteToImage(imageBytes);
+                    images.add(bitImage);
+                    pictureCursor.moveToNext();
+                }
+                pictureCursor.close();
+                Incident incident = new Incident(name, queryResults.getString(descriptionIndex), weather, images);
+                allIncidents.add(incident);
+                queryResults.moveToNext();
+            }
+            queryResults.close();
+        }catch(Exception e){
+            System.out.println("Error getting all incidents: " + e.getMessage());
+        }
+        return allIncidents;
+    }
+
+    private byte[] imageToByte(Bitmap image){
+        ByteArrayOutputStream oStream = new ByteArrayOutputStream();
+        image.compress(Bitmap.CompressFormat.PNG, 50, oStream);
+        byte[] imageBytes = oStream.toByteArray();
+        return imageBytes;
+    }
+
+    private Bitmap byteToImage(byte[] bImage){
+        Bitmap bitmapImage = BitmapFactory.decodeByteArray(bImage, 0, bImage.length);
+        return bitmapImage;
+    }
+
+    public void dropAllTables(){
+        String dropIncidentTable = "DROP TABLE " + INCIDENT_TABLE;
+        String dropImageTable = "DROP TABLE " + PICTURE_TABLE;
+        try{
+            db.execSQL(dropIncidentTable);
+            System.out.println("Dropped incident table");
+            db.execSQL(dropImageTable);
+            System.out.println("Dropped picture table");
+        }catch(Exception e){
+            System.out.println("Error dropping tables: " + e.getMessage());
+        }
+    }
+
+    public int getImageCount(){
+        int size = -1;
+        Cursor countCursor;
+        try{
+            countCursor = db.rawQuery("SELECT * FROM image_table;", null);
+            size = countCursor.getCount();
+        }catch(Exception e){
+            System.out.println("Error getting count: " + e.getMessage());
+        }
+        return size;
+    }
+
+    public void removeAllRows(){
+        String removeIncidentRows = String.format("DELETE FROM %1$s;", INCIDENT_TABLE);
+        String removePictureRows = String.format("DELETE FROM %1$s;", PICTURE_TABLE);
+        try{
+            db.execSQL(removeIncidentRows);
+            db.execSQL(removePictureRows);
+        }catch(Exception e){
+            System.out.println("Error deleting all rows: " + e.getMessage());
```

Comment:
Same as above. It's important to make sure that the exception is handled in the right place. This might be the correct place, I'm not sure, depends on what this function is being used for.